modules/apps/portal-crypto-hash/portal-crypto-hash-impl/src/main/java/com/liferay/portal/crypto/hash/internal/CryptoHashVerifierImpl.java

```diff
@@ -42,9 +42,19 @@ public boolean verify(
 		for (CryptoHashVerificationContext cryptoHashVerificationContext :
 				cryptoHashVerificationContexts) {
 
-			CryptoHashProvider cryptoHashProvider =
-				_cryptoHashProviderRegistry.getCryptoHashProvider(
-					cryptoHashVerificationContext.getCryptoHashProviderName());
+			CryptoHashProvider cryptoHashProvider;
+
+			try {
+				cryptoHashProvider =
+					_cryptoHashProviderFactoryRegistry.getCryptoHashProvider(
+						cryptoHashVerificationContext.
+							getCryptoHashProviderFactoryName(),
+						cryptoHashVerificationContext.
+							getCryptoHashProviderProperties());
+			}
+			catch (Exception exception) {
+				throw new CryptoHashException(exception);
+			}
```

Comment:
why are we adding this try/catch now?

Reply:
Because during provider creation phase, some providers may throw exception, for example: MD.
Here is a commit I attached to the last, to move done the try catch from impl to provider creation:
https://github.com/liferay-appsec/liferay-portal/pull/294/commits/08eccb58f6175770962126b9604faeb398e54df9

If this is not necessary, we can simply remove this commit.